server/src/main/java/io/unitycatalog/server/persist/DeltaCommitRepository.java

```diff
@@ -9,25 +9,48 @@
 import io.unitycatalog.server.model.DeltaCommitMetadataProperties;
 import io.unitycatalog.server.model.DeltaMetadata;
 import io.unitycatalog.server.model.TableType;
+import io.unitycatalog.server.persist.dao.ColumnInfoDAO;
 import io.unitycatalog.server.persist.dao.DeltaCommitDAO;
+import io.unitycatalog.server.persist.dao.PropertyDAO;
 import io.unitycatalog.server.persist.dao.TableInfoDAO;
 import io.unitycatalog.server.persist.utils.FileOperations;
 import io.unitycatalog.server.persist.utils.TransactionManager;
+import io.unitycatalog.server.utils.Constants;
+import io.unitycatalog.server.utils.IdentityUtils;
 import io.unitycatalog.server.utils.ServerProperties;
 import io.unitycatalog.server.utils.ValidationUtils;
 import java.util.Comparator;
+import java.util.Date;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.UUID;
 import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.query.NativeQuery;
 import org.hibernate.query.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** Repository for managing coordinated commits for managed Delta tables in Unity Catalog. */
 public class DeltaCommitRepository {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(DeltaCommitRepository.class);
+
+  /**
+   * The maximum number of commits allowed per table before backfilling is required. TODO: turn this
+   * into a configurable server property.
+   */
+  private static final int MAX_NUM_COMMITS_PER_TABLE = 50;
```

Comment:
lets keep this fairly small. say 10. because number of commits across all the table can get large. so limit of 10 means commit table size 10x # tables.

Reply:
yes. thank you for adding this.